x-pack/plugins/alerting/server/saved_objects/migrations.ts

```diff
@@ -869,3 +904,36 @@ function pipeMigrations(...migrations: AlertMigration[]): AlertMigration {
   return (doc: SavedObjectUnsanitizedDoc<RawRule>) =>
     migrations.reduce((migratedDoc, nextMigration) => nextMigration(migratedDoc), doc);
 }
+
+/**
+ * This creates a migration map that applies search source migrations to legacy es query rules
+ */
+function getEsQueryAlertSearchSourceMigrations(
+  searchSourceMigrations: MigrateFunctionsObject
+): MigrateFunctionsObject {
+  return mapValues<MigrateFunctionsObject, MigrateFunction>(
+    searchSourceMigrations,
+    (
+        migrateSerializedSearchSourceFields: MigrateFunction<SerializedSearchSourceFields>
+      ): MigrateFunction =>
+      (state) => {
+        const _state = state as { attributes: RawRule };
+
+        const serializedSearchSource = _state.attributes.params.searchConfiguration;
```

Comment:
**question:** should we check the rule type before applying modifications? using `isEsQueryRuleType` on `_state.attributes.alertTypeId`.

Reply:
We do not need it, since `serializedSearchSource` is used only as `searchConfiguration` rule param only in `esQuery` alerts.

Reply:
When future alerts will use `searchSource`, we will need it to change it.

Reply:
I tend to agree with Mike, for two reasons:
1.  I don't think we should rely on `searchConfiguration` only being used by ESQuery - it's too implicit. There's nothing in the code preventing other rule types from using that key.
2. We do not want to decrypt every single rule - that impacts the performance of the migration path. If we can reduce the number of rules we need to decrypt by adding this check ,then it's worth doing.

Reply:
We'll never remember to change it, if some future rule type uses this param, for some other purpose.  :-)

So I think we should add the check, OR make it more general and obvious.  For instance, change `getEsQueryAlertSearchSourceMigrations()` to `getSearchSourceMigrations()` and pass along an additional array of rule types to test against.  I doubt this will be the last rule type to use searchSource and need this.

I suppose we could also consider having some "standardized" parameter names, like one for search source.  Seems kinda magical though ... seems like it could cause surprises in the future.

Reply:
Check added.